addons/point_of_sale/static/src/js/models.js

```diff
@@ -909,29 +909,29 @@ exports.PosModel = Backbone.Model.extend({
             partnerModel.loaded(this, partners);
         }
     },
-    loadProductsFromOffset: async function(offset) {
-        let ProductIds = [];
+    // Load the products into the `db` and return the amount loaded
+    loadProductsFromPage: async function(page) {
         let product_model = _.find(this.models, (model) => model.model === 'product.product');
-        ProductIds = await this.rpc({
-            model: 'product.product',
-            method: 'search',
-            args: [product_model.domain(this)],
+        const products = await this.rpc({
+            model: 'pos.config',
+            method: 'get_limited_products_loading',
+            args: [[this.config_id]],
             kwargs: {
-                offset: this.env.pos.config.limited_products_amount * offset,
-                limit: this.env.pos.config.limited_products_amount,
+                fields: product_model.fields,
+                page,
             },
-            context: this.env.session.user_context,
+            context: { ...this.session.user_context, ...product_model.context() },
         });
-        return ProductIds;
+        product_model.loaded(this, products);
+        return products.length
     },
     loadProductsBackground: async function() {
-        let offset = 0
-        let ProductIds = [];
+        let page = 1
+        let productAmount = 0;
         do {
-            ProductIds = await this.loadProductsFromOffset(offset)
-            this.env.pos._addProducts(ProductIds);
-            offset += 1;
-        } while(ProductIds.length);
+            productAmount = await this.loadProductsFromPage(page)
+            page += 1;
+        } while(productAmount);
```

Comment:
What do you think of using `while (productAmount == self.config.limited_products_amount)`. Basically, if the last page you fetched is less than `self.config.limited_products_amount`, then it is the final page. But while it is the same count as the `limited_products_amount`, it should try to fetch the next page.

Reply:
letsgo